drivers/SmartThings/matter-lock/src/init.lua

```diff
@@ -587,14 +625,15 @@ local function device_added(driver, device)
   if #eps == 0 then
     if device:supports_capability_by_id(capabilities.tamperAlert.ID) then
       device.log.debug("Device does not support lockCodes. Switching profile.")
-      local power_source_eps = device:get_endpoints(clusters.PowerSource.ID)
-      local battery_feature_eps = device:get_endpoints(clusters.PowerSource.ID, {feature_bitmap = clusters.PowerSource.types.PowerSourceFeature.BATTERY})
       local profile_name = "lock-without-codes"
-      if #power_source_eps == 0 then
+      device:set_field(PROFILE_BASE_NAME, profile_name, {persist = true})
+      local battery_feature_eps = device:get_endpoints(clusters.PowerSource.ID, {feature_bitmap = clusters.PowerSource.types.PowerSourceFeature.BATTERY})
+      if #battery_feature_eps == 0 then
         profile_name = profile_name .. "-nobattery"
-      elseif #battery_feature_eps == 0 then
+      elseif not device:get_field(SUPPORT_BATTERY_PERCENTAGE) then
```

Comment:
Similar to above, since this is being handled in the `added` handler, I wouldn't expect this value to have been updated yet since it relies on a read request that is sent during the initial onboarding. This will always be false then, and the device would always join to a `-batteryLevel` profile. I would suggest again that we count on the AttributeList handler to determine what battery support needs to be added to the profile, as mentioned above. I think this would have the added benefit of centralizing this logic in once place too!

Reply:
Like in the other comment, I changed this to only update the profile if the battery feature is not available!